.chloggen/fix_container-parser-configurable-batch-size.yaml

```
@@ -0,0 +1,29 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```

```suggestion
change_type: breaking
```

I think we should mark it as a breaking change.

The previous behavior with `max_batch_size: 0` was the same as with `max_batch_size: 1`. This behavior changes now. As unlikely as it is for a user to have used `max_batch_size: 0` in their configuration, I'd rather be on the safe side and mention this change in the "breaking changes" section.